### PR TITLE
kv indexer: add separator to start key when matching ranges

### PR DESCRIPTION
**Pull request by melekes**
to avoid including false positives

Refs #2908

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: state/txindex/kv/kv.go [hunk 1]
```diff
@@ -429,11 +419,36 @@ func extractValueFromKey(key []byte) string {
...
+		result.Index,
+	))
 }
 
 func keyForHeight(result *types.TxResult) []byte {
-	return []byte(fmt.Sprintf("%s/%d/%d/%d", types.TxHeightKey, result.Height, result.Height, result.Index))
+	return []byte(fmt.Sprintf("%s/%d/%d/%d",
+		types.TxHeightKey,
+		result.Height,
+		result.Height,
```
Comment (ebuchman) [inline state/txindex/kv/kv.go]: Is height supposed to be here twice?

File: state/txindex/kv/kv.go [hunk 2]
```diff
@@ -429,11 +419,36 @@ func extractValueFromKey(key []byte) string {
...
+	if height > 0 {
+		return startKey(c.Tag, c.Operand, height)
+	}
+	return startKey(c.Tag, c.Operand)
+}
+
+func startKey(fields ...interface{}) []byte {
+	var b bytes.Buffer
+	for _, f := range fields {
+		b.Write([]byte(fmt.Sprintf("%v", f) + tagKeySeparator))
```
Comment (ebuchman) [inline state/txindex/kv/kv.go]: Why the `+` when we're already using Sprintf?

File: state/txindex/kv/kv.go (hunk 1 above)
Comment (melekes) [reply state/txindex/kv/kv.go]: Yes. It's sort of a hack.

File: state/txindex/kv/kv.go (hunk 2 above)
Comment (melekes) [reply state/txindex/kv/kv.go]: You mean?

```suggestion
		b.Write([]byte(fmt.Sprintf("%v%v", f, tagKeySeparator)))
```

File: state/txindex/kv/kv.go (hunk 1 above)
Comment (Liamsi) [reply state/txindex/kv/kv.go]: Should we add a comment here about the rationale of this hack?

**Comment by melekes** [comment]
@Liamsi thanks! added a test case

**Pull request merged**